tsdb/chunks/chunks.go

```diff
@@ -774,3 +770,22 @@ func sequenceFiles(dir string) ([]string, error) {
 	}
 	return res, nil
 }
+
+// ChunkSizeInBytes calculates the total size in bytes of a given chunk.
+// It includes the length of the chunk's data, the length of the variable-length
+// integer representation of this data length, the chunk encoding size, and the
+// CRC32 checksum size.
+// The varintBuffer parameter is used to avoid allocating a new buffer on every
+// call, optimizing performance by reusing the provided buffer for encoding the
+// variable-length integer. The function returns the total size as an integer.
+// A new buffer of MaxChunkLengthFieldSize can be created with: make([]byte, MaxChunkLengthFieldSize).
+func ChunkSizeInBytes(chunk chunkenc.Chunk, varintBuffer []byte) int {
+	chunkDataLength := len(chunk.Bytes())
+	chunkDataLengthUvarintLength := binary.PutUvarint(varintBuffer, uint64(chunkDataLength))
+	chunkSize := 0
+	chunkSize += chunkDataLengthUvarintLength // The data length is a variable length field.
+	chunkSize += ChunkEncodingSize            // The chunk encoding.
+	chunkSize += chunkDataLength              // The data itself.
+	chunkSize += crc32.Size                   // The 4 bytes of crc32.
+	return chunkSize
+}
```

Comment:
Nice, amazing it calculates accurate uvarint!

However, I wonder if this function could be less verbose -- especially it's simple enough that such lengthy commentary will be prone to aging vs the benefit. Also the key aspect is that it's encoded size. 

Notably we comment about the details 3 times, in function comment, in verbose variable names and in field comments in function body. Perhaps one of those is enough, wdyt?

What about:

```suggestion
// EncodedSizeInBytes calculates the total size in bytes of a given chunk, when
// encoded by the chunks.Writer. It requires a bytes buffer of at least MaxChunkLengthFieldSize
// size to avoid allocations.
func EncodedSizeInBytes(chunk chunkenc.Chunk, buf []byte) (size int64) {
	size += int64(len(chunk.Bytes())) // The data itself.
	size += binary.PutUvarint(buf, uint64(size)) // The data length is a variable length field.
	size += ChunkEncodingSize // The chunk encoding.
	size += crc32.Size  // The 4 bytes of crc32.
	return int64(size)
}
```

Reply:
Thank you! We addressed this :)